ansible_collections/arista/avd/roles/eos_designs/python_modules/connected_endpoints/utils.py

```diff
@@ -97,7 +97,7 @@ def _get_short_esi(self, adapter: dict, channel_group_id: int, short_esi: str =
         """
         Return short_esi for one adapter
         """
-        if len(set(adapter["switches"])) < 2:
+        if len(set(adapter["switches"])) < 2 and not self.shared_utils.overlay_evpn and not self.shared_utils.overlay_vtep:
```

Comment:
This logic will not work correctly if only one of the condtions are false.
```suggestion
        if len(set(adapter["switches"])) < 2 or not self.shared_utils.overlay_evpn or not self.shared_utils.overlay_vtep:
```

Please add a test case in molecule to make sure we see the problem and then see that your solution solves it.

Reply:
Fixed.